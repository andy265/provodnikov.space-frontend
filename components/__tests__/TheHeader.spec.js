const { createRenderer } = require('vue-server-renderer')
const { createLocalVue, shallow } = require('vue-test-utils')
const TheHeader = require('~/components/TheHeader')

describe('TheHeader.vue', () => {
  it('has same HTML structure', () => {
    let Component = Object.assign({}, TheHeader, { components: {} })
    let wrapper = shallow(Component)
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('open subscribe modal on subscribe click', () => {
    let wrapper = shallow(TheHeader, {
      localVue: createLocalVue(),
      mocks: {
        $router: {
          push: jest.fn()
        },
        $route: {
          path: '/page'
        }
      }
    })
    wrapper.vm.onSubscribeClick()
    expect(wrapper.vm.$router.push.mock.calls[0][0]).toEqual('/page#subscribe')
  })
})
