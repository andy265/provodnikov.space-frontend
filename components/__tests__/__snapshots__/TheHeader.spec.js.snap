// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader.vue has same HTML structure 1`] = `
<header class="the-header">
    <TheHeaderLogo class="the-header__logo"></TheHeaderLogo>
    <div class="the-header__inner">
        <AppTitle to="/" tag="h1" theme="main" class="the-header__title">Андрей Проводников / Блог</AppTitle>
        <div>
            <AppButton>подписаться</AppButton>
        </div>
    </div>
</header>
`;
